src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,314 @@
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="report">Report type to retrieve from the context.</param>
+        /// <returns>
+        /// The instance of DiagnosticsData.
+        /// </returns>
+        /// <feature>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </feature>
+        public DiagnosticsData GetReport(ReportType report)
+        {
+            var ret = NativeApi.GetReport(_handle, (NativeApi.ReportType)report, out IntPtr data);
+            if (ret != NativeApi.DiagnosticsError.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return new DiagnosticsData(data);
+        }
+
+        /// <summary>
+        /// Stores context type.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public ContextType Type { get; internal set; }
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose API for closing the internal resources.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (_handle != IntPtr.Zero)
+                {
+                    NativeApi.Destroy(_handle);
+                    _handle = IntPtr.Zero;
+                }
+                _disposed = true;
+            }
+        }
+    }
+
+    /// <summary>
+    /// The DiagnosticsData class is used for reading data (either from report or client).
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public class DiagnosticsData : IDisposable
+    {
+        private IntPtr _handle = IntPtr.Zero;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The DiagnosticsData class constructor.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="handle">Diagnostics data handle.</param>
+        internal DiagnosticsData(IntPtr handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Finalizes an instance of the DiagnosticsData class.
+        /// </summary>
+        ~DiagnosticsData()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Reads given number of bytes from diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="size">Number of bytes to read.</param>
+        /// <param name="timeout">Timeout [ms] for reading requested number of bytes.</param>
+        /// <returns>
+        /// An array of bytes read.
+        /// </returns>
+        public unsafe byte[] Read(int size, int timeout)
```

Comment:
Avoid to open `unsafe` method as public. Use unsafe block in the method body or make and wrap another unsafe private method.